hvpy/utils.py

```diff
@@ -127,3 +129,26 @@ def create_events(events: List[Union[EventType, str, tuple]]) -> str:
             raise ValueError(f"{event} is not a EventType or str or two-length tuple")
     # Strips the final comma
     return constructed_events[:-1]
+
+
+def save_file(data: bytearray, filename: str, overwrite: Optional[bool] = False) -> None:
+    """
+    Saves a file to the specified path.
+
+    Parameters
+    ----------
+    data
+        The data to save.
+    filename
+        The path to save the file to.
+    overwrite
+        Whether to overwrite the file if it already exists.
+        Default is `False`, optional.
+    """
+    if Path(filename).exists():
+        if overwrite:
+            Path(filename).write_bytes(data)
+        else:
+            raise ValueError(f"{filename} already exists, use overwrite=True to overwrite")
+    else:
+        Path(filename).write_bytes(data)
```

Comment:
This should be condensed, you should do the logic check once, raise the error or then write the file.